### PR TITLE
add global-real-auto-save-mode

### PR DESCRIPTION
**Pull request by DivineDominion**
I'm not an experiences Emacs package developer, so a review and feedback would be much appreciated!

With the global mode enabled, one can disable `real-auto-save-mode` on a per-buffer or per-major-mode basis. I found this easier for my setup than enabling auto saving for every conceivable mode.

I tested it this way:

```
;; init.el
(require 'real-auto-save)
(setq real-auto-save-interval 10) ;; in seconds
(defun turn-on-real-auto-save () (real-auto-save-mode 1))
(defun turn-off-real-auto-save () (real-auto-save-mode -1))
(define-globalized-minor-mode global-real-auto-save-mode
  real-auto-save-mode turn-on-real-auto-save)
(global-real-auto-save-mode 1)
(add-hook 'message-mode-hook #'turn-off-real-auto-save)
```

Auto-saving in .txt files works, auto-saving in message drafts is disabled. Manually disabling this in a buffer via `M-: (real-auto-save-mode -1)` works, too, and doesn't affect the next file you visit (i.e. auto-saving is still enabled by default).

### CODE REVIEW

**Comment by ChillarAnand** [comment]
Any suggestions? @conao3

**Comment by conao3** [comment]
I think this is a good idea, but `real-auto-save-initialize` is better than `turn-on-real-auto-save` because `turn-on-*` is not well known prefix.
And `turn-off-real-auto-save` is not needed I think.

**Comment by DivineDominion** [comment]
Thanks for the feedback! I got the inspiration for `turn-on-*` and `turn-off-*` from existing packages. Most are not interactive, but if you hit `M-: (turn-on TAB` (`eval-expression`, then type `(turn-on`, then TAB to get auto-completion) you'll see quite a few packages use this. (`turn-off-*` is far less common, that's right)

In Magit's `git-commit.el`, for example, you'll find calls to `(turn-on-flyspell)` and `(turn-on-autofill)`. I just picked a random package from my currently frontmost buffer's mode list.

I think we can live without these functions, but I also think they are useful API. Users can then declare

```elisp
(add-hook 'message-mode-hook #'turn-off-real-auto-save)
```

... and don't need to create lambdas or whatever.

Please note I'm not sure if this is still considered weird. I just picked this up during the past weeks and liked the convention.

**Comment by conao3** [comment]
OK, thanks. I'm not professional too, but I follow package-lint suggestion.
Please install package-lint and see warnings.
```
 real-auto-save.el   104   1 error           "turn-on-real-auto-save" doesn't start with package's prefix "real-auto-save". (emacs-lisp-package)
 real-auto-save.el   104     info            All variables and subroutines might as well have a documentation string (emacs-lisp-checkdoc)
 real-auto-save.el   105   1 error           "turn-off-real-auto-save" doesn't start with package's prefix "real-auto-save". (emacs-lisp-package)
 real-auto-save.el   105     info            All variables and subroutines might as well have a documentation string (emacs-lisp-checkdoc)
```
So, I suggest to use `real-auto-save-*` prefix.
And I think, the add-hook you suggestion is not needed. Please define `real-auto-save-initialize` as like below.
```emacs-lisp
(defun real-auto-save-initialize ()
  (when buffer-file-name (real-auto-save-mode))
```

**Comment by conao3** [comment]
I usually use `(add-hook 'find-file-hook 'real-auto-save-mode)`, and it is fine I think, but the global-minor-mode you suggest is the better idea.

**Comment by DivineDominion** [comment]
Whoa that year went by fast. I addressed some of your comments @conao3 but didn't recall if you want the `-turn-off` function to be removed completely (then you cannot opt-out of global mode easily), and if you want `-turn-on` to be renamed to `-initialize`.

Grepping through the emacs lisp sources, I only found a handful of uses of the `turn-on-` prefix, notably `turn-on-auto-revert-mode`. In my own emacs config, I only find 22 completions for `turn-on-`. I have 32 matches for `-initialize$`, but most of these do something else than my proposed functions, like perform setup work. I found `yas-initialize` to be like the turn-on-convention, but it toggles the global mode (which makes more sense for YASnippets). Btw I found 5 matches for `-turn-on$` suffix.

TL;DR: the '-initialize` suffix sounds like it'd be misleading. 

----

I'm not a fan of the `-turn-on` and `-turn-off` suffix because it reads weird. I'm fine with removing the helpers completely and just add the global minor mode, then add the `turn-off-` helper in my own config for consistency and that's it :)

**Review by conao3 — APPROVED** [review]
I was organizing the PR list and came back to this PR.
This change looks good.
